### PR TITLE
Move UpdateModelConfigDefaultValues method to state.Model receiver.

### PR DESCRIPTION
**Pull request by ExternalReality**
## Description of change

CAAS Refactor. Move `UpdateModelConfigDefaultValues` from `state.State` to `state.Model`

### CODE REVIEW

**Comment by ExternalReality** [comment]
!!build!!

File: apiserver/common/modelmanagerinterface.go [hunk 1]
```diff
@@ -116,7 +117,13 @@ func (st modelManagerStateShim) NewModel(args state.ModelArgs) (Model, ModelMana
 	if err != nil {
 		return nil, nil, err
 	}
-	return modelShim{m}, modelManagerStateShim{otherState, st.pool}, nil
+	otherModel, err := otherState.Model()
```
Comment (wallyworld) [inline apiserver/common/modelmanagerinterface.go]: The model result from NewModel() above is the otherModel - why make the same API call again here?

File: apiserver/common/modelmanagerinterface.go [hunk 2]
```diff
@@ -101,13 +101,14 @@ var _ ModelManagerBackend = (*modelManagerStateShim)(nil)
...
+	model *state.Model
+	pool  *state.StatePool
 }
 
 // NewModelManagerBackend returns a modelManagerStateShim wrapping the passed
 // state, which implements ModelManagerBackend.
-func NewModelManagerBackend(st *state.State, pool *state.StatePool) ModelManagerBackend {
-	return modelManagerStateShim{st, pool}
+func NewModelManagerBackend(st *state.State, m *state.Model, pool *state.StatePool) ModelManagerBackend {
+	return modelManagerStateShim{st, m, pool}
```
Comment (wallyworld) [inline apiserver/common/modelmanagerinterface.go]: We could avoid the churn in all these constructors which take both a state and model by:
- change now to take just model arg
- add a State() api to Model (just temporary)
- use the State() api to get state inside ModelManagerBackend

then when state is no longer required, we can drop the API off Model

File: state/model.go [hunk 3]
```diff
@@ -1292,6 +1292,10 @@ func (m *Model) getEntityRefs() (*modelEntityRefsDoc, error) {
 	return &doc, nil
 }
 
+func (model *Model) State() *State {
```
Comment (wallyworld) [inline state/model.go]: Add a comment and also a TODO that is temporary and will go once refactoring is complete

**Comment by ExternalReality** [comment]
$$MERGE$$

**Pull request merged**